enumerables.rb

```diff
@@ -0,0 +1,154 @@
+# Enumerable methods
+
+module Enumerable
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+
+  # my_each
+
+  def my_each
+    return to_enum unless block_given?
+    return to_a if self.class == Range
+
+    length.times { |i| yield(self[i]) }
+    self
+  end
+
+  # end of my_each
+
+  # my_each_with_index
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    length.times { |i| yield(self[i], i) }
+    self
+  end
+
+  # end of my_each_with_index
+
+  # my_select
+
+  def my_select
+    return to_enum unless block_given?
+
+    arr = []
+    my_each { |i| arr.push(i) if yield(i) }
+    arr
+  end
+
+  # end of my_select
+
+  # my_all?
+
+  def my_all?(arg = nil)
+    if_all = true
+    my_each do |i|
+      if block_given? || arg.nil?
```

Comment:
- [ ] my_all is not having the expected behavior when a block is given.
Ex: 
`[1,2,3,4,5].my_all? { |n| n > 1 }`
`=> true`
The above should return false
But it returns true